runtime/autoload/remote/host.vim

```diff
@@ -204,37 +210,43 @@ function! s:RequirePythonHost(name)
         \ ' -c "import sys; sys.stdout.write(str(sys.version_info[0]) + '.
         \ '\".\" + str(sys.version_info[1]))"'
 
-  let supported = ['2.6', '2.7']
+  let supported = (ver == 2) ?
+        \ ['2.6', '2.7'] : ['3.3', '3.4', '3.5']
 
-  " To load the python host a python executable must be available
-  if exists('g:python_host_prog')
-        \ && executable(g:python_host_prog)
-        \ && index(supported, system(g:python_host_prog.get_version)) >= 0
-    let python_host_prog = g:python_host_prog
+  " To load the Python/Python3 host a Python/Python3 executable must be
+  " available
+  if has_key(g:, host_var)
+        \ && executable(g:[host_var])
+        \ && index(supported, system(g:[host_var].get_version)) >= 0
+    let python_host_prog = g:[host_var]
   elseif executable('python')
         \ && index(supported, system('python'.get_version)) >= 0
     let python_host_prog = 'python'
-  elseif executable('python2')
+  elseif ver == 3 && executable('python3')
+        \ && index(supported, system('python3'.get_version)) >= 0
+    " In some distros, python2 is the default python command
+    let python_host_prog = 'python3'
+  elseif ver == 2 && executable('python2')
         \ && index(supported, system('python2'.get_version)) >= 0
-    " In some distros, python3 is the default python
+    " In some distros, python3 is the default python command
```

Comment:
People aren't necessarily using a "distro", so perhaps it could be:

On some platforms, `python2` is symlinked to `python`.